### ISSUE TITLE
Ceph volumes cannot be attached to the pods after 1.12

### ISSUE THREAD

**Issue by grebennikov** [juju-solutions/bundle-canonical-kubernetes#662]
Deploying with the same bundle before/after switching to 1.12 by default.
Bundle: https://pastebin.ubuntu.com/p/TY4YW5myBh/
Error: https://pastebin.ubuntu.com/p/5rwM7hCfq2/

Prior to the version update there was no storage class created by default by kubernetes-master charm, as well as there was not option "default-storage" available (kubernetes 1.11 was used).
I was creating it manually and everything worked perfectly. Now I'm getting 2 cni-XXX storage classes, one of them marked as default. I'm creating Ceph pool, the volumes get created in it and marked as Bound in "get pvc" output, however they cannot be used by pods.

Also, unless I set "allow-privileged=true" to kubernetes-master config, the units cannot succeed with an error while installing addons: https://pastebin.ubuntu.com/p/M8V4FFgCkp/
If I tried to execute same command manually:
```sudo /snap/cdk-addons/445/kubectl apply -f /root/snap/cdk-addons/445/addons --recursive -l cdk-addons=true --prune=true --force```
 - the install command exits with "Not allowed by cluster policy"

**Comment by xannz** [comment]
Running into the same problem after following these [instructions](https://github.com/juju-solutions/kubernetes-docs/blob/master/pages/k8s/ceph.md). K8s cluster is also version 1.12.

**Comment by Cynerva** [comment]
I've opened three PRs to fix the originally reported issue:
https://github.com/juju-solutions/cdk-addons/pull/59
https://github.com/juju-solutions/cdk-addons/pull/60
https://github.com/juju-solutions/cdk-addons/pull/61

The key bit from the error output was this:
```
2018-10-09 03:50:03.268166 7f75987a0d40  0 librados: client.kubernetes authentication error (1) Operation not permitted
```

The charms don't create a `client.kubernetes` user in our Ceph cluster. The template we pull from upstream was updated to specify it, and we left it in: https://github.com/ceph/ceph-csi/commit/b0e68a52e0e533523cd9bf7e2f06aa894ffeafdc#diff-592b6313f1ac4f9644734ca5bc1a0710R27

The PRs fix it by forcing it back to the `admin` user.

**Comment by Cynerva** [comment]
Please note that, while testing this fix, I encountered other issues with Ceph that are equally problematic. I recommend subscribing to them as well: https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/668 https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/669

**Comment by Cynerva** [comment]
Fixed by https://github.com/juju-solutions/cdk-addons/pull/59, which has just been released via cdk-addons on channel 1.12/stable. New deployments should work immediately, and existing deployments should update within the next 24 hours.

> Also, unless I set "allow-privileged=true" to kubernetes-master config, the units cannot succeed with an error while installing addons
